Core.Collectors/IO/CloudQueueMsiWrapper.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Queue;
+using Newtonsoft.Json;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.CloudMine.Core.Collectors.IO
+{
+    public class CloudQueueMsiWrapper : IQueue
+    {
+        private readonly CloudQueue queue;
+        private readonly string queueName;
+        private readonly string storageAccountNameEnvironmentVariable;
+
+        public CloudQueueMsiWrapper(CloudQueue queue, string storageAccountNameEnvironmentVariable)
+        {
+            this.queue = queue;
+            this.queueName = queue.Name;
+            this.storageAccountNameEnvironmentVariable = storageAccountNameEnvironmentVariable;
+        }
+
+        public Task PutObjectAsJsonStringAsync(object obj)
+        {
+            string message = JsonConvert.SerializeObject(obj);
+            return this.PutMessageAsync(message);
+        }
+
+        public async Task PutMessageAsync(string message)
+        {
+            CloudQueue queue = await AzureHelpers.GetStorageQueueUsingMsiAsync(queueName, storageAccountNameEnvironmentVariable).ConfigureAwait(false);
+            await queue.AddMessageAsync(new CloudQueueMessage(message)).ConfigureAwait(false);
+        }
+
+        public async Task PutMessageAsync(string message, TimeSpan timeToLive)
```

Comment:
Should we combine these two implementations by using a default `timeToLive` value?

Reply:
What would be a default `timeToLive` value?

Reply:
I think you can default it to null and this will be the same as not passing it in, but this would let you get rid of the extra function declaration. This is no big deal though.

Reply:
All these methods are consistent with other queue classes like InMemoryQueue, CloudWrapperQueue which in turn inherit from IQueue. I like the way they are structured as each of this grouped method has a designated functionality. I prefer to leave it as is if that is okay